### ISSUE TITLE
Value(bar) to monitor the progress of scan.

### ISSUE THREAD

**Issue by MarkHe1222** [anarchy9388/scan#16]

**Comment by MarkHe1222** [comment]
command:

lglf131:/tmp/mark # python3 scan3.py scan -f chunkids -s /tmp/free_blocks_with_history_e6b1e40d-adfa-4bb8-9ed3-4e3e4b87e6d8_2019-07-15-07-39-26
scan3.py v0.3                 Started 2019-07-15 09:39:43

Log file will be saved in: /tmp/scan_busy_blocks_2019-07-15-09-39-43.log
Result will be saved in dir: /tmp/block_scan_res
Total: 12711 candidate blocks need to pre-check.


Execution Time:
1.0 minutes 19 seconds


result:

lglf131:/tmp/block_scan_res # ls
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6073346400_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6140455200_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6075583360_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6142692160_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6077820320_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6144929120_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6080057280_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6147166080_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6082294240_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6149403040_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6084531200_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6151640000_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6086768160_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6153876960_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6089005120_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6156113920_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6091242080_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6158350880_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6093479040_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6160587840_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6095716000_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6162824800_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6097952960_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6165061760_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6100189920_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6167298720_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6102426880_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6169535680_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6104663840_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6171772640_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6106900800_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6174009600_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6109137760_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6176246560_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6111374720_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6178483520_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6113611680_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6180720480_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6115848640_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6182957440_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6118085600_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6185194400_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6120322560_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6187431360_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6122559520_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6189668320_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6124796480_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6191905280_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6127033440_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6194142240_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6129270400_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6196379200_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6131507360_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6198616160_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6133744320_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6200853120_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6135981280_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6203090080_2236960
2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6138218240_2236960  2ae9e5af-4473-4606-b039-353bd87401f6_11841294-5909-4bcf-833c-d97de5ddce0e_0039_6205327040_2236960

**Comment by anarchy9388** [comment]
@MarkHe1222 

A progress bar may greatly impact performance are you sure this is needed? 
I could do an updating counter that shows how many file parts we have processed and the total to be processed?

**Comment by MarkHe1222** [comment]
just pushed the latest version which is runnalbe.

**Comment by MarkHe1222** [comment]
@anarchy9388  I think we no need the progress bar, we just need to find a way to monitor the progress . 
so there is question here. for some candidate which should treated as finished during the pre-checking stage
and for another candidate blocks will keep moving to the next stage to parse the write block byte by byte

**Comment by MarkHe1222** [comment]
I think we can just print the count to the screen to show the progress of the script

example:
     total candidate: 233333 , remaining: 11111  

what do you think?

**Comment by MarkHe1222** [comment]
BTW:

the latest version:

command:
```
lglf131:/tmp/mark # vim scan3.py
lglf131:/tmp/mark # python3 scan3.py scan -f chunkids -s /tmp/free_blocks_with_history_e6b1e40d-adfa-4bb8-9ed3-4e3e4b87e6d8_2019-07-31-03-34-53
scan3.py v0.8                 Started 2019-07-31 09:21:14

Log file will be saved in: /tmp/scan_busy_blocks_2019-07-31-09-21-14.log
Result will be saved in dir: /tmp/block_scan_results
Total: 21210 candidate blocks need to pre-check.
```

result in /tmp/block_scan_results:
```
lglf131:/tmp/block_scan_results # ls -alh | tail
-rw-r--r-- 1 root root  22K Jul 31 09:22 f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3735519713_21683
-rw-r--r-- 1 root root  22K Jul 31 09:22 f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3735541396_21683
-rw-r--r-- 1 root root  22K Jul 31 09:22 f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3735563079_21683
-rw-r--r-- 1 root root  22K Jul 31 09:22 f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3735584762_21683
-rw-r--r-- 1 root root  22K Jul 31 09:22 f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3735606445_21747
-rw-r--r-- 1 root root  22K Jul 31 09:22 f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3735628192_21683
-rw-r--r-- 1 root root  22K Jul 31 09:22 f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3735649875_21683
-rw-r--r-- 1 root root  22K Jul 31 09:22 f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3735671558_21683
-rw-r--r-- 1 root root  22K Jul 31 09:22 f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3735693241_21683
-rw-r--r-- 1 root root 8.1K Jul 31 09:22 f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3735714924_8276
```

log:
```
lglf131:/tmp # grep "Target write block saved in" /tmp/scan_busy_blocks_2019-07-31-09-21-14.log | head
2019-07-31 09:22:20,694 scan3.py[line:257] INFO Target write block saved in: /tmp/block_scan_results/f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3724538400_6171
2019-07-31 09:22:20,715 scan3.py[line:257] INFO Target write block saved in: /tmp/block_scan_results/f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3724544571_21731
2019-07-31 09:22:20,723 scan3.py[line:257] INFO Target write block saved in: /tmp/block_scan_results/f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3724566302_21683
2019-07-31 09:22:20,731 scan3.py[line:257] INFO Target write block saved in: /tmp/block_scan_results/f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3724587985_21683
2019-07-31 09:22:20,733 scan3.py[line:257] INFO Target write block saved in: /tmp/block_scan_results/f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3724609668_21683
2019-07-31 09:22:20,735 scan3.py[line:257] INFO Target write block saved in: /tmp/block_scan_results/f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3724631351_21683
2019-07-31 09:22:20,738 scan3.py[line:257] INFO Target write block saved in: /tmp/block_scan_results/f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3724653034_21683
2019-07-31 09:22:20,740 scan3.py[line:257] INFO Target write block saved in: /tmp/block_scan_results/f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3724674717_21683
2019-07-31 09:22:20,743 scan3.py[line:257] INFO Target write block saved in: /tmp/block_scan_results/f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3724696400_21683
2019-07-31 09:22:20,745 scan3.py[line:257] INFO Target write block saved in: /tmp/block_scan_results/f25a9f6f-6ad1-4c6a-945f-dcf66a41648a_faac27e5-07cc-4ac5-ae13-08393918c8f8_0316_3724718083_21683
```

**Comment by anarchy9388** [comment]
@MarkHe1222 

Made progress counters for the script... Merged the branch from ISSUE-16 into 0.8.
```
scan3dev.py v0.8                 Started 2019-07-31 19:02:38

Log file will be saved in: /tmp/scan_busy_blocks_2019-07-31-19-02-38.log
Result will be saved in dir: /tmp/block_scan_results
Total: 21210 candidate blocks need to pre-check.

Total parts to pre-check: 1280    Processed Parts: 1218    Received: 1    Target Blocks Found: 521    Running Candidate Block Scanner processes:62    Bytes Processed: 9.52 GB
```

One thing I did notice is the script is hanging at certain points on pre-check chunks but not sure why yet...
